source/extensions/retry/host/omit_host_metadata/omit_host_metadata.cc

```diff
@@ -0,0 +1,37 @@
+#include "extensions/retry/host/omit_host_metadata/omit_host_metadata.h"
+
+#include "common/config/metadata.h"
+#include "common/config/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Retry {
+namespace Host {
+
+bool OmitHostsRetryPredicate::shouldSelectAnotherHost(const Upstream::Host& host) {
+  // Check if host doesn't have any metadata.
+  if (host.metadata()->filter_metadata().find(Envoy::Config::MetadataFilters::get().ENVOY_LB) ==
+      host.metadata()->filter_metadata().end()) {
+    return false;
+  }
+
+  // Check if there's no metadata match criteria configured.
+  const auto& filter_it = metadata_match_criteria_.filter_metadata().find(
```

Comment:
how about we validate this at construction and throw an `EnvoyException` if its not properly configured? That should let us clean up this error handling here

Reply:
Done